Sources/TuistGraph/Models/ProjectOption.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+
+/// Additional options related to the `Project`
+public enum ProjectOption: Codable {
+    /// Text settings to override user ones for current project
+    case textSettings(TextSettings)
```

Comment:
Is there any reason to not use associated values here as well as for the ProjectDescription?

Reply:
Yeah, thought about it. 
But imo it's better to have model for `TextSettings` inside graph. 
In this case instead of doing `case let .textSettgins(usesTabs, indentWidth, tabWidth, wrapsLines)` each time, we can just extract `TextSettings` and simply use it independently. Also it's easier to test/mock.

Reply:
I agree with @DimaMishchenko here as it'll be easier to work with and it's just for internal usage